src/gsy_e/models/strategy/predefined_load.py

```diff
@@ -25,9 +25,56 @@
 from gsy_e.gsy_e_core.exceptions import GSyException
 from gsy_e.gsy_e_core.global_objects_singleton import global_objects
 from gsy_e.gsy_e_core.util import should_read_profile_from_db
+from gsy_e.models.state import LoadState
 from gsy_e.models.strategy.load_hours import LoadHoursStrategy
 
 
+class DefinedLoadEnergyParameters:
+    def __init__(self, daily_load_profile=None, daily_load_profile_uuid: str = None):
+        self.profile_uuid = daily_load_profile_uuid
+        self._load_profile_W = None
+        self._load_profile_kWh = {}
+        self.state = LoadState()
+
+        if should_read_profile_from_db(daily_load_profile_uuid):
+            self._load_profile_input = None
+        else:
+            self._load_profile_input = daily_load_profile
+
+    def event_activate_energy(self):
+        """
+        Runs on activate event.
+        :return: None
+        """
+        self._read_or_rotate_profiles()
+
+    def reconfigure(self, **kwargs):
+        if key_in_dict_and_not_none(kwargs, "daily_load_profile"):
+            self._load_profile_input = kwargs["daily_load_profile"]
+            self._read_or_rotate_profiles(reconfigure=True)
+
+    def _read_or_rotate_profiles(self, reconfigure=False):
```

Comment:
Nit, this seems to be used as an interface rather than an internal method

Reply:
I have converted the method to public in a followup commit.